JIT/EE interface cleanup

jkotas opened the pull request:
Delete methods on JIT/EE interface that are no longer used in CoreCLR

jkotas committed:
JIT/EE interface cleanup

Delete methods on JIT/EE interface that are no longer used in CoreCLR

jkotas commented:
cc @dotnet/jit-contrib

jkotas committed:
Fix build break

jkotas committed:
Remove workaround for getLocationOfThisType method

jkotas committed:
Fix build break

jkotas committed:
Formatting patch

jkotas committed:
Merge branch 'master' into jit-ee-cleanup

jkotas committed:
Keep comment

jkotas committed:
Fix crossgen2

AndyAyersMS reviewed (commented):
Would be good to verify this is zero-diff.

Seems like we could get rid of `GTF_INX_REFARR_LAYOUT` entirely.

jkotas committed:
Delete GTF_INX_REFARR_LAYOUT

jkotas committed:
Add comment

AndyAyersMS commented:
Superpmi not happy.

BruceForstall commented:
The SuperPMI unit test finally found what it was designed to find. Nice.

jkotas committed:
Hardcode verificationNeeded to false

jkotas committed:
Merge branch 'master' into jit-ee-cleanup

jkotas committed:
Fix superpmi

AndyAyersMS commented on src/coreclr/src/jit/compiler.h:
```diff
@@ -9240,8 +9240,10 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-    BOOL tiVerificationNeeded;
+    // CoreCLR does not ever run IL verification. Compile out the verifier from the JIT by making this a constant.
+    // TODO: Delete the verifier from the JIT?
+    // BOOL tiVerificationNeeded;
```
Yeah, I think we should do this, but it doesn't have to be done now.

AndyAyersMS reviewed (approved):
Thanks, Jan.

jkotas committed:
Revert "Delete GTF_INX_REFARR_LAYOUT"

This reverts commit c73f640a5cc613f7216beb148f7884946c07ab10.

jkotas committed:
Add comment

jkotas commented:
I have reverted the `GTF_INX_REFARR_LAYOUT` layout removal. It was causing asm diffs. I have added TODO comment for it instead so that it can be dealt with separately - will open an issue.

Example of asm diff:

Baseline:
```
mov rcx, qword ptr [0x1ce08]  // object (TYPE_HANDLE)
call qword ptr [0x10c8]       // GET_RUNTIME_TYPE_HANDLE (HELPER)
cmp qword ptr [rbp + 160], rax
```

With `GTF_INX_REFARR_LAYOUT` removed:
```
mov rbp, qword ptr [rcx + 160]
mov rcx, qword ptr [0x1ce08]  // object (TYPE_HANDLE)
call qword ptr [0x10c8]       // GET_RUNTIME_TYPE_HANDLE (HELPER)
cmp rbp, rax
```

My guess is that that there is a place in register allocator (have not traced down where exactly) that compares all flags without interpreting them.

jkotas committed:
Added links to issues

jkotas committed:
Delete getModuleNativeEntryPointRange

jkotas committed:
Delete unused file

jkotas committed:
Delete .gitmirror files

jkotas commented:
```
src/installer/pkg/projects/Directory.Build.props(4,3): error : System.AggregateException: One or more errors occurred. (Could not find a part of the path '/tmp/NuGetScratch/eeea79e456214a75b7f5887557384111/136e41645a9a49b0860273810a1411c4.proj.nuget.dgspec.json'.)
 ---> System.IO.DirectoryNotFoundException: Could not find a part of the path '/tmp/NuGetScratch/eeea79e456214a75b7f5887557384111/136e41645a9a49b0860273810a1411c4.proj.nuget.dgspec.json'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode)
   at NuGet.ProjectModel.DependencyGraphSpec.Save(String path)
   at NuGet.Commands.NoOpRestoreUtilities.PersistDGSpecFile(DependencyGraphSpec spec, String dgPath, ILogger log)
   at NuGet.Commands.RestoreCommand.EvaluateCacheFile()
   at NuGet.Commands.RestoreCommand.ExecuteAsync(CancellationToken token)
   at NuGet.Commands.RestoreRunner.ExecuteAsync(RestoreSummaryRequest summaryRequest, CancellationToken token)
   at NuGet.Commands.RestoreRunner.CompleteTaskAsync(List`1 restoreTasks)
   at NuGet.Commands.RestoreRunner.RunWithoutCommit(IEnumerable`1 restoreRequests, RestoreArgs restoreContext)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Microsoft.Build.NuGetSdkResolver.NuGetSdkResolver.NuGetAbstraction.GetSdkResult(SdkReference sdk, Object nuGetVersion, SdkResolverContext context, SdkResultFactory factory)
```

jkotas commented:
Known issue: https://github.com/dotnet/runtime/issues/32638
